### PR TITLE
Add general readme about devfile.

### PR DESCRIPTION
**Pull request by mshaposhnik**
Signed-off-by: Max Shaposhnik <mshaposh@redhat.com>

### What does this PR do?
Adds general HOWTO doc page about devfile && make it deployable to docs site

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/12732

### CODE REVIEW

**Review by l0rd — APPROVED** [review]

File: wsmaster/che-core-api-devfile/README.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
+
+
+### Introduction
+Previously, the  two kind of recipes was available to bootstrap a cloud developer workspace and make it portable: [Chefile](https://www.eclipse.org/che/docs/chefile.html)
```
Comment (l0rd) [inline wsmaster/che-core-api-devfile/README.md]: s/the  two/two/
s/was available/were available/
s/and make it/and to make it/

File: wsmaster/che-core-api-devfile/README.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
+
+
+### Introduction
+Previously, the  two kind of recipes was available to bootstrap a cloud developer workspace and make it portable: [Chefile](https://www.eclipse.org/che/docs/chefile.html) 
+and [Factories](https://www.eclipse.org/che/docs/factories-getting-started.html#try-a-factory).
+As a continuation of this, the brand new `devfile` format was introduced, which unites the forming simplicity and the high variety of different tools available to develop a container based application.
```
Comment (l0rd) [inline wsmaster/che-core-api-devfile/README.md]: s/which unites the forming simplicity and the high variety/which combines simplicity and support for high variety/

File: wsmaster/che-core-api-devfile/README.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+### What the devfile consists of
+The minimal devfile sufficient to run a workspace from it, consists of the following parts:
+ - Specification version
+ - Name
+ - Tools set
```
Comment (l0rd) [inline wsmaster/che-core-api-devfile/README.md]: s/Tools set/A list of tools: the development tools and user runtimes/

File: wsmaster/che-core-api-devfile/README.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+As a continuation of this, the brand new `devfile` format was introduced, which unites the forming simplicity and the high variety of different tools available to develop a container based application.
+
+### What the devfile consists of
+The minimal devfile sufficient to run a workspace from it, consists of the following parts:
+ - Specification version
+ - Name
+ - Tools set 
+ 
+To get more functional workspace, the following parts can be added:
+ - Project(s)
```
Comment (l0rd) [inline wsmaster/che-core-api-devfile/README.md]: s/Project(s)/A list of projects: the source code repositories/

File: wsmaster/che-core-api-devfile/README.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+ - Command(s)
```
Comment (l0rd) [inline wsmaster/che-core-api-devfile/README.md]: s/Command(s)/A list of commands: actions to manage the workspace components like running the dev tools, starting the runtime environments etc.../

File: wsmaster/che-core-api-devfile/README.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+Also, it is possible to execute devfile by constructing the factory with the URL to it's raw content, for example, 
+```
+https://<your-che-host>/f?url=https://pastebin.com/raw/ux6iCGaW
+``` 
+or sending a devfile to a dedicated REST API using curl/swagger, which will create new workspace and return it's configuration: 
+```
+curl -X POST  -H "Authorization: <TOKEN>" -H "Content-Type: application/yaml" -d <devlile_content> https://<your-che-host>/api/devfile
+``` 
+
+If you're an user of `chectl` tool, is is also possible to execute workspace from devfile, using `workspace:start` command
```
Comment (l0rd) [inline wsmaster/che-core-api-devfile/README.md]: s/an user/a user/
s/is is also/it is also/

File: wsmaster/che-core-api-devfile/README.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+parameter as follows:
+```
+chectl workspace:start --devfile=.devfile
+```` 
+Please note that currently this way only works for the local devfiles.
+
+
+ 
+### Supported Tool types
+There is currently four types of tools supported. There is two simpler types, such as `cheEditor` and `chePlugin` and
```
Comment (l0rd) [inline wsmaster/che-core-api-devfile/README.md]: s/There is/There are/

File: wsmaster/che-core-api-devfile/README.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+  tools:
+    - name: mysql
+      type: kubernetes
+      local: petclinic.yaml
+      selector:
+        app.kubernetes.io/name: mysql
+        app.kubernetes.io/component: database
+        app.kubernetes.io/part-of: petclinic
+```
+Contents of the `local` file is currently read _ONLY_ if the devfile and local file both placed in the same GitHub repository.
```
Comment (skabashnyuk) [inline wsmaster/che-core-api-devfile/README.md]: same GitHub repository - > same public GitHub repository

File: wsmaster/che-core-api-devfile/README.md [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+    - name: mysql
+      type: kubernetes
+      local: petclinic.yaml
+      selector:
+        app.kubernetes.io/name: mysql
+        app.kubernetes.io/component: database
+        app.kubernetes.io/part-of: petclinic
+```
+Contents of the `local` file is currently read _ONLY_ if the devfile and local file both placed in the same GitHub repository. 
+So, alternatively, if you need to post devfile with such tools to REST API, contents of K8S/Openshift list can be put directly into devfile using `localContent` field:
```
Comment (skabashnyuk) [inline wsmaster/che-core-api-devfile/README.md]: put directly-> embedded?

File: wsmaster/che-core-api-devfile/README.md [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+     actions:
+       - type: exec
+         tool: mysql
+         command: mvn clean
+         workdir: /projects/spring-petclinic
+```
+
+### Live working examples
+
+    //TODO
```
Comment (skabashnyuk) [inline wsmaster/che-core-api-devfile/README.md:184]: What the plan here?

**Review by skabashnyuk — COMMENTED** [review]
overall looks good

File: wsmaster/che-core-api-devfile/README.md (hunk 10 above)
Comment (mshaposhnik) [reply wsmaster/che-core-api-devfile/README.md:184]: I've prepared couple of working examples (web-nodejs-sample, web-java-spring-petclinic) but che.openshift.io is not able to run any of them. (for me seems Che 7 workspaces not working here at all).
So they still currlently in my own repo, and after we will be able to run it we will move somewhere into public example repos and provide links here.

Comment (l0rd) [reply wsmaster/che-core-api-devfile/README.md:184]: @mshaposhnik go ahead even if they don't work on che.openshift.io. There is a known issue that prevent che 7 workspace to run on OSO starter cluster (cf. https://che.statuspage.io)

Comment (mshaposhnik) [reply wsmaster/che-core-api-devfile/README.md:184]: Ok so there is 3 repo with exampes:
1) https://github.com/mshaposhnik/web-nodejs-sample  very simple  H/W example;
2) https://github.com/mshaposhnik/web-java-spring-petclinic  port to Che 7using devfile
3) https://github.com/mshaposhnik/hello-world-frontend-plugin  fork of one of Theia plugin with devfile.

Any ideas where to host them? Into `che-samples` maybe but indicate they're an Che 7 samples?

Comment (l0rd) [reply wsmaster/che-core-api-devfile/README.md:184]: What about `github.com/redhat-developers/devfile/samples`?

**Review by skabashnyuk — APPROVED** [review]

File: wsmaster/che-core-api-devfile/README.md (hunk 10 above)
Comment (mshaposhnik) [reply wsmaster/che-core-api-devfile/README.md:184]: Ok, lets merge this one, and after that i prepare examples & add them as separate PR.

**Pull request merged**